src/main/java/org/osc/manager/rest/server/api/SecurityApis.java

```diff
@@ -0,0 +1,173 @@
+ * Copyright (c) Intel Corporation
+ * Copyright (c) 2017
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.osc.manager.rest.server.api;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.stream.Collectors;
+
+import javax.persistence.EntityManager;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.apache.log4j.Logger;
+import org.osc.manager.ism.api.IsmSecurityGroupApi;
+import org.osc.manager.ism.api.IsmSecurityGroupInterfaceApi;
+import org.osc.manager.ism.entities.SecurityGroupEntity;
+import org.osc.manager.ism.entities.SecurityGroupInterfaceEntity;
+import org.osc.manager.rest.server.SecurityManagerServerRestConstants;
+import org.osc.sdk.manager.element.ManagerSecurityGroupElement;
+import org.osc.sdk.manager.element.ManagerSecurityGroupInterfaceElement;
+import org.osc.sdk.manager.element.SecurityGroupInterfaceElement;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.transaction.control.TransactionControl;
+
+@Component(service = SecurityApis.class)
+@Path(SecurityManagerServerRestConstants.SERVER_API_PATH_PREFIX + "/securitygroups")
+@Consumes({ MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML })
+@Produces({ MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML })
+public class SecurityApis {
+    private static Logger LOG = Logger.getLogger(SecurityApis.class);
+    private EntityManager em;
+    private TransactionControl txControl;
+    private IsmSecurityGroupApi sgApi;
+    private IsmSecurityGroupInterfaceApi sgiApi;
+
+    public void init(EntityManager em, TransactionControl txControl) throws Exception {
+        this.em = em;
+        this.txControl = txControl;
+        this.sgApi = new IsmSecurityGroupApi(null, txControl, em);
+        this.sgiApi = new IsmSecurityGroupInterfaceApi(null, null, txControl, em);
+    }
+
+    @POST
+    public String createSecurityGroup(SecurityGroupEntity entity) throws Exception {
+        LOG.info(String.format("Creating security group  with name %s", "" + entity.getName()));
+        return this.sgApi.createSecurityGroup(entity.getName(), null, null);
```

Comment:
Also ensure to provide the iscId from the entity as well instead of null.

Reply:
As per the plugin code, oscid is not part of the entity structure and also it is not used in the plugin code. @larkinscarvalho - can you comment?
Nevertheless, I will add this parameter in REST layer so that user can key in if needed.